### ISSUE TITLE
Data loss caused by header overwrite

### ISSUE THREAD

**Issue by Mubelotix** [neonmoe/minreq#56]
Minreq is currently unable to handle headers appearing multiple times in the response. 
Only the last value is stored, as it is inserted over the previous values.
https://github.com/neonmoe/minreq/blob/efbaf757515cd77070de5cb3e0a9ece49e20df92/src/response.rs#L303

**Comment by neonmoe** [comment]
Yep, this is planned for [3.0](https://github.com/neonmoe/minreq#planned-for-300), since it would require changing the Response struct's public API.

**Comment by Mubelotix** [comment]
Great! I need a quick fix so I will do it myself but it would be great if that could benefit to other users too. I could do a PR. How would you prefer to solve this? Using a custom `HeaderMap` [like reqwest did](https://docs.rs/reqwest/0.11.3/reqwest/header/struct.HeaderMap.html) or using a `HashMap<String, Vec<String>>`?

**Comment by neonmoe** [comment]
I haven't given it much thought. Reqwest's solution certainly makes for a clean API, but I fear that would require quite a bit of code, and I'd rather keep minreq as simple as possible. I think a `HashMap<String, Vec<String>>` would be good if combined with a convenience function for Response, e.g. `Response::get_header(key: &str) -> Option<&str>`, which would just return the first header, if any.

**Comment by neonmoe** [comment]
Fixed in d1279702cc2e64fcecfc92f0a15e41ee43615f9b! Ended up changing the `HashMap<String, String>` into `Vec<(String, String)>` as that's what the headers actually are. For more "accelerated" lookups, if the server is responding with >100 headers for example, I'd recommend making your own `HashMap` from the `Vec` that fits your needs.
